File: androidApp/src/main/java/com/example/grup/android/ui/Colors.kt
package com.example.grup.android.ui

import androidx.compose.runtime.*
import androidx.compose.ui.graphics.Color

val dark_grey = Color(0xFF1F1F1F)
val grey = Color(0xFF292929)
val light_grey = Color(0xFF3D3D3D)
val white = Color(0xFFFFFFFF)
val off_white = Color(0xFFF5F5F5)
val green = Color(0xFF65B540)
val red = Color(0xFFEF1A1A)
val red_error = Color(0xffff0033)

class AppColors(
    primary: Color,
    secondary: Color,
    onPrimary: Color,
    onSecondary: Color,
    confirm: Color,
    deny: Color,
    caption: Color,
    error: Color
) {
    var primary by mutableStateOf(primary)
        private set
    var secondary by mutableStateOf(secondary)
        private set
    var onPrimary by mutableStateOf(onPrimary)
        private set
    var onSecondary by mutableStateOf(onSecondary)
        private set
    var confirm by mutableStateOf(confirm)
        private set
    var deny by mutableStateOf(deny)
        private set
    var caption by mutableStateOf(caption)
        private set
    var error by mutableStateOf(error)
        private set
    fun copy(
        primary: Color = this.primary,
        secondary: Color = this.secondary,
        onPrimary: Color = this.onPrimary,
        onSecondary: Color = this.onSecondary,
        confirm: Color = this.confirm,
        deny: Color = this.deny,
        caption: Color = this.caption,
        error: Color = this.error,
    ): AppColors = AppColors(
        primary,
        secondary,
        onPrimary,
        onSecondary,
        confirm,
        deny,
        caption,
        error,
    )

    fun updateColorsFrom(other: AppColors) {
        primary = other.primary
        secondary = other.secondary
        onPrimary = other.onPrimary
        onSecondary = other.onSecondary
        confirm = other.confirm
        deny = other.deny
        caption = other.caption
        error = other.error
    }
}

fun appColors(
    primary: Color = dark_grey,
    secondary: Color = grey,
    onPrimary: Color = off_white,
    onSecondary: Color = white,
    confirm: Color = green,
    deny: Color = red,
    caption: Color = light_grey,
    error: Color = red_error
): AppColors = AppColors(
    primary = primary,
    secondary = secondary,
    onPrimary = onPrimary,
    onSecondary = onSecondary,
    confirm = confirm,
    deny = deny,
    caption = caption,
    error = error
)

internal val LocalColors = staticCompositionLocalOf{ appColors() }

File: iosApp/iosApp/ui/Colors.swift
//
//  Colors.swift
//  iosApp
//
//  Created by Justin Xu on 4/11/23.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import SwiftUI

extension Color {
    init(hex: UInt, alpha: Double = 1) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xff) / 255,
            green: Double((hex >> 08) & 0xff) / 255,
            blue: Double((hex >> 00) & 0xff) / 255,
            opacity: alpha
        )
    }
}

struct Colors {
    static let dark_grey = Color(hex: 0xFF1F1F1F)
    static let grey = Color(hex: 0xFF292929)
    static let light_grey = Color(hex: 0xFF3D3D3D)
    static let white = Color(hex: 0xFFFFFFFF)
    static let off_white = Color(hex: 0xBBF5F5F4)
    static let green = Color(hex: 0xFF65B540)
    static let red = Color(hex: 0xFFEF1A1A)
    static let red_error = Color(hex: 0xffff0033)
    
    let primary: Color = dark_grey
    let secondary: Color = grey
    let onPrimary: Color = off_white
    let onSecondary: Color = white
    let confirm: Color = green
    let deny: Color = red
    let caption: Color = light_grey
    let error: Color = red_error
}
